### DISCUSSION TITLE
Multifield subquery support

### DISCUSSION THREAD

**Discussion by paymog** [prisma/prisma#20488] [Q&A]
## Problem

I have the following sql query which I would like to run with Prisma, but I can't figure out how (maybe I'm not being creative enough!)

```sql
SELECT s.*
FROM "Stream" s
WHERE (project_id, name, version) IN (
    SELECT project_id, name, MAX(version)
    FROM "Stream"
    GROUP BY project_id, name
);
```

For context, I have a table, `Stream` which defines a compound primary key

```sql
create unique index "Stream_project_id_name_version_key"
    on "Stream" (project_id, name, version);
```

I'd like to select all rows which have the highest `version` for a given `project_id`, `name` pair. As far as I can tell, the above query works but I'm not sure how to run it in Prisma.

I know I can use `$queryRaw` but [the docs](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#executeraw) suggest raising an issue for queries that require `queryRaw`.

## Suggested solution

It would be great if Prisma supported subqueries for compound fields like in the above SQL query.

## Alternatives

Using `$queryRaw` works

**Comment by ludralph** [comment]
Hello @paymog 👋 

We do have a feature request to support subqueries in Prisma. You can leave a 👍 and add your use case to help our engineers prioritize it. As for the query you shared, using `$queryRaw` looks like the best approach. You can also leverage [SafeQL](https://safeql.dev/compatibility/prisma.html) to write raw queries in a type-safe way.

- https://github.com/prisma/prisma/issues/5617

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
